tests/test-polynomial-matrix.C

```diff
@@ -0,0 +1,425 @@
+/*
+ * Copyright (C) 2013  Pascal Giorgi
+ *
+ * Written by Pascal Giorgi <pascal.giorgi@lirmm.fr>
+ *
+ * ========LICENCE========
+ * This file is part of the library LinBox.
+ *
+ * LinBox is free software: you can redistribute it and/or modify
+ * it under the terms of the  GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ * ========LICENCE========
+ */
+
+
+
+
+#include <iostream>
+#include <iomanip>
+using namespace std;
+
+#include <linbox/ring/modular.h>
+#include <linbox/randiter/random-prime.h>
+#include <linbox/randiter/random-fftprime.h>
+#include <givaro/zring.h>
+#include <recint/rint.h>
+#include <linbox/matrix/matrix-domain.h>
+#include <linbox/util/commentator.h>
+#include <linbox/util/timer.h>
+#include <linbox/matrix/polynomial-matrix.h>
+#include <linbox/algorithms/polynomial-matrix/polynomial-matrix-domain.h>
+#include <linbox/algorithms/polynomial-matrix/matpoly-mult-naive.h>
+#include <linbox/algorithms/polynomial-matrix/matpoly-mult-kara.h>
+
+using namespace LinBox;
+
+template<typename Rand, typename Matrix>
+void randomMatPol(Rand& r,  Matrix& A){
+	for(size_t i=0;i<A.size()*A.rowdim()*A.coldim();i++)
+		r.random(A.getPointer()[i]);
+}
+
+
+template<typename T>
+void test(T& x){
+    x.ref(0,0)++;
+}
+
+template <typename Field, PMType T1, PMType T2>
+bool operator!= (const PolynomialMatrix<Field, T1>& A,
```

Comment:
Same here: can't we use the methode `areEqual` of the matrix domain?

Reply:
This raises the question of compliancy to the dense matrix model. Looking deeper in the `areEqual` code of the matrix domain, this implies to provide row and column iterators to the polynomial matrix class. I'am not sure we want this for the current polynomial matrix classes. 

This question of compliancy with the dense matrix model is also raised with the question of getEntry` and `refEntry` below. Here again, this polynomial matrix class are not designed  to be used as a classic dense matrix. If one to do so, he would have to construct a dense matrix over the adequate polynomial ring.

Overall, we may want to support both compatibility but this would imply several change in the polynomial matrix API and in some matrix domain. I propose to postpone this to a future enhancement.

Reply:
Ok then.